demo/api/src/products/generated/product.resolver.ts

```diff
@@ -32,19 +31,22 @@ export class ProductResolver {
 
     @Query(() => Product)
     @AffectedEntity(Product)
+    @RequiredPermission(["createProduct", "updateProduct", "deleteProduct"], { skipScopeCheck: true })
     async product(@Args("id", { type: () => ID }) id: string): Promise<Product> {
```

Comment:
for single-resolver I combined all permissions used for create, update and delete as this mutations usually also need to fetch a single entity. single-resolver should have an extra option to define permissions.

Reply:
IMO this should be products, same as in list. A user may be allowed to read a product, even if they are not allowed to modify it.

Reply:
So if I understand you correctly I should also add the list-permission? Alright

Reply:
yes, but call the permission `read` instead of `list`